### ISSUE TITLE
Screen reader support for form components

### ISSUE THREAD

**Issue by geospatialem** [Esri/calcite-components#4600]
### Summary

Navigating through the form, some components labels are not being read back to the screen reader. In these cases the label is not present and is skipped.

### Actual Behavior

Some form components labels are not being read back by JAWS. The items of question include:

  - _2. Where did you hear about Calcite components?_
      -   `calcite-select`
  - _3. Will you be using Calcite components in an upcoming project?_
      -   `calcite-radio-button-group`
  - _4. How many components will you be using?_
      - `calcite-slider`
  -   _6. How useful is the Calcite design system?_
      - `calcite-radio-group` 
  - _7. What is the most important aspect(s) of a component?_
      - `calcite-checkbox`
  - _9. How many stars would you give Calcite components?_
      - `calcite-rating`    

### Expected Behavior

Each label should be read back by the screen reader.

### Reproduction Sample

https://codepen.io/geospatialem/pen/JjpJdwg

### Reproduction Steps

1. Open the [Codepen example](https://codepen.io/geospatialem/pen/JjpJdwg)
2. Open JAWS and navigate the form
3. The following components with `calcite-labels` are not read back to the user via JAWS:
    - _2. Where did you hear about Calcite components?_
        -   `calcite-select`
    - _3. Will you be using Calcite components in an upcoming project?_
        -   `calcite-radio-button-group`
    - _4. How many components will you be using?_
        - `calcite-slider`
    -   _6. How useful is the Calcite design system?_
        - `calcite-radio-group` 
    - _7. What is the most important aspect(s) of a component?_
        - `calcite-checkbox`
    - _9. How many stars would you give Calcite components?_
        - `calcite-rating`    

### Reproduction Version

beta.82

### Relevant Info

Verified in JAWS
cc #4388

Notes:
Can be mitigated by adding a `<fieldset>` and `<legend>`, such as:

```html
<li>
  <fieldset>
    <legend>
	    <calcite-label>How many stars would you give Calcite components?
    </legend>
	  <calcite-rating value="1"></calcite-rating>
  </calcite-label>
  </fieldset>
</li>
```

**Comment by driskull** [comment]
Can we update that codepen to use the latest release? Maybe the `next` cdn version? Its currently using beta 62.

The codepen should also put the components within the `calcite-label` or use a `for` attribute to link the component and the label. I see the ones in question are not inside the `calcite-label` nor do they have a `for` attribute.

**Comment by geospatialem** [comment]
Great idea! 🧠 💡  

Updated the Codepen using `next` and added additions for `calcite-label` with the `for` attribute.

**Comment by geospatialem** [comment]
@driskull unfortunately yes. Apologies for not including that above. I noticed it while putting together samples for the audit in the latest release as well.

There is a workaround using the fieldset and legend, but as-is the labels aren't read back by JAWS.

**Comment by driskull** [comment]
Thanks for testing!

Just want to make sure that the code is correct and the issue still persists. It won't read the label if the component isn't inside a calcite-label or doesn't have the for property.

@geospatialem can you remove the for property from components that are nested inside the label? It shouldn't have both. It should be one or the other but not both.

**Comment by geospatialem** [comment]
Will be adding related issues from the a11y audit that will reference this issue once posted.

**Comment by geospatialem** [comment]
Closing in favor of #5627
